src/Scene/Scene.js

```diff
@@ -180,15 +159,49 @@ define('Scene/Scene', [
         this.gfxEngine.update();
     };
 
-    Scene.prototype.wait = function(timeWait) {
+    Scene.prototype.notifyChange = function() {
+        if(!this.forceUpdate || ! this.needsUpdate) {
+            this.needsUpdate = true;
+            this.forceUpdate = true;
+            requestAnimationFrame(this.step.bind(this));
+        }
+    };
+
+    Scene.prototype.runCommands = function() {
+        var quadtree = this.layers[0].node.tiles;
+        var process = this.layers[0].process;
+
+        this.browserScene.browse(quadtree,this.currentCamera(), process, SUBDIVIDE);
+        var allDone = this.managerCommand.fillCommandPool();
+        if(allDone) {
+            this.browserScene.browse(quadtree,this.currentCamera(), process, CLEAN);
+            this.viewerDiv.dispatchEvent(event);
+            this.needsUpdate = false;
+        }
+    };
+
+    Scene.prototype.step = function(timestamp) {
 
-        var waitTime = timeWait ? timeWait: 20;
+        if(this.forceUpdate || this.needsUpdate && (this.managerCommand.executedCommands >= 16 || (timestamp - this.lastTimestamp) > 200 && this.managerCommand.executedCommands !== 0)) {
```

Comment:
This big conditional statement, mixing && and ||, needs to be cleaned/simplified.
At the very least, introduce some simple boolean:
```js
var atLeast16commandsExecutedSinceLastRender = this.managerCommand.executedCommands >= 16;
var timeout5fps = (timestamp - this.lastTimestamp) > 200; // not sure why you want at least 1 command executed here?
```